### ISSUE TITLE
Consider refactoring the x86 hwintrinsic flags

### ISSUE THREAD

**Issue by tannergooding** [dotnet/coreclr#16656]
Many of the x86 Hardware Intrinsic flags have negative naming conventions, such as: `NoContainment`, `NoJumpTableIMM`, and `NoRMWSemantics`.

We should consider refactoring these into the positive case so that the resulting code is easier to read.

Having helper functions (https://github.com/dotnet/coreclr/issues/16594) may be another way to keep the code more readable without also having to reverse the flags to match the common case (positive), rather than the uncommon case (negative).

**Comment by tannergooding** [comment]
CC. @CarolEidt, @fiigii

**Comment by fiigii** [comment]
NoContainment, NoJumpTableIMM, and NoCodegen is fine, just a few intrinsics need them.

**Comment by fiigii** [comment]
When we design a flag, I think that should be used as less as possible.

**Comment by fiigii** [comment]
@tannergooding You have exposed certain convenient functions for these flags, shall we close this?

**Comment by tannergooding** [comment]
I think @CarolEidt was the one who had remaining feedback on the flags having "negative naming" which made reading them in source code confusing.

**Comment by fiigii** [comment]
> having "negative naming" which made reading them in source code confusing.

Yes, and the convenient functions already help a lot.

**Comment by tannergooding** [comment]
Right, I don't disagree. This was left open due to the listed preference that the flags still be named "positively" as well: https://github.com/dotnet/coreclr/pull/18255#discussion_r192567416

I'll let @CarolEidt close if she thinks that we don't need to rename the root flags given that we have the positively named helper functions now.

**Comment by CarolEidt** [comment]
> When we design a flag, I think that should be used as less as possible.

If a flag must be checked, it doesn't generally matter perf-wise whether you are checking to see if it is set or unset, and I reiterate that negative flags are undesirable. If we *must* have the flags as-is, we should strive to rename them, e.g. `NoContainment` becomes `RegOperandsOnly`.

**Comment by fiigii** [comment]
> If a flag must be checked, it doesn't generally matter perf-wise whether you are checking to see if it is set or unset, and I reiterate that negative flags are undesirable. 

From the perf perspective, YES. However, I meant that "used as less as possible" can make the code more maintainable. For example, if a flag is required by 99% intrinsic, it means the other 1% is special and we just need to care the 1%, rather than exploding the concerns to other "normal" ones.

**Comment by fiigii** [comment]
> rather than exploding the concerns to other "normal" ones.

That helps when we add a new intrinsic into the table. We just need to consider what features the new one has, instead of what features that other ones have but the new one not.
